### PR TITLE
info level log for pipeline abort

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description

Add a summarize stack log util, use it in Pipeline to summarize abort errors at an info level

### CODE REVIEW

File: services/pipeline/src/main/java/org/hyperledger/besu/services/pipeline/Pipeline.java [hunk 1]
```diff
@@ -176,8 +176,7 @@ private Future<?> runWithErrorHandling(final ExecutorService executorService, fi
                 || t instanceof AsyncOperationException) {
               LOG.trace("Unhandled exception in pipeline. Aborting.", t);
             } else {
-              LOG.info("Unexpected exception in pipeline. Aborting.");
-              LOG.debug("Unexpected exception in pipeline. Aborting.", t);
+              LOG.info("Unexpected exception in pipeline. Aborting.", t);
```
Comment (macfarla) [inline services/pipeline/src/main/java/org/hyperledger/besu/services/pipeline/Pipeline.java:179]: // log the stacktrace at info to help troubleshoot sync stuck issues

**Review by macfarla — APPROVED** [review]
suggestion to add a comment to make it obvious we want this stacktrace at info level

File: services/pipeline/src/main/java/org/hyperledger/besu/services/pipeline/Pipeline.java (hunk 1 above)
Comment (garyschulte) [reply services/pipeline/src/main/java/org/hyperledger/besu/services/pipeline/Pipeline.java:179]: maybe it should be at warn actually

File: util/src/main/java/org/hyperledger/besu/util/log/LogUtil.java [hunk 2]
```diff
@@ -36,4 +39,34 @@ public static void throttledLog(
...
+   * @param throwable exception to summarize
+   * @param namespace namespace to summarize to
+   * @return summary of the StackTrace
+   */
+  public static String summarizeStackTrace(
+      final String contextMessage, final Throwable throwable, final String namespace) {
+    StackTraceElement[] stackTraceElements = throwable.getStackTrace();
+
+    List<String> stackTraceSummary = new ArrayList<>();
+    for (StackTraceElement element : stackTraceElements) {
```
[the remark this answers is not in the document]
Comment (garyschulte) [reply util/src/main/java/org/hyperledger/besu/util/log/LogUtil.java:68]: added a max depth of 20, since this is supposed to be a summary 👍

**Pull request merged**